akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/TestKitSettings.scala

```diff
@@ -67,11 +67,11 @@ final class TestKitSettings(val config: Config) {
    * Scala API: Scale the `duration` with the configured `TestTimeFactor`
    */
   def dilated(duration: FiniteDuration): FiniteDuration =
-    (duration * TestTimeFactor).asInstanceOf[FiniteDuration]
+    Duration.fromNanos((duration.toNanos * TestTimeFactor + 0.5).toLong)
 
   /**
    * Java API: Scale the `duration` with the configured `TestTimeFactor`
    */
   def dilated(duration: java.time.Duration): java.time.Duration =
-    java.time.Duration.ofMillis((duration.toMillis * TestTimeFactor).toLong)
+    java.time.Duration.ofMillis((duration.toMillis * TestTimeFactor + 0.5).toLong)
```

Comment:
in the Scala variant we use nanos, perhaps delegate so that we only have one implementation
```
import akka.util.JavaDurationConverters._
def dilated(duration: java.time.Duration): java.time.Duration =
  dilated(duration.asScala).asJava
```

Reply:
Sounds good to me!